starlite/plugins/sql_alchemy/config.py

```diff
@@ -236,6 +236,29 @@ def validate_before_send_handler(  # pylint: disable=no-self-argument
         """
         return AsyncCallable(value)  # type: ignore[arg-type]
 
+    @root_validator
+    def check_connection_string_or_engine_instance(  # pylint: disable=no-self-argument
+        cls, values: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Either `connection_string` or `engine_instance` must be specified,
+        and not both.
+
+        Args:
+            values: Field values, after validation.
+
+        Returns:
+            Field values.
+        """
+        connection_string = values.get("connection_string")
+        engine_instance = values.get("engine_instance")
+        if connection_string is None and engine_instance is None:
+            message = "One of 'connection_string' or 'engine_instance' must be provided."
```

Comment:
I would simply raise here. Then change elif to if. Same goes after

Reply:
yeh agree